src/domain/data/index.ts

```diff
@@ -0,0 +1,44 @@
+export class DataProvider {
+  private readonly encoder = new TextEncoder();
+  constructor(
+    private readonly textLen: number,
+    private readonly patternLen: number
+  ) {}
+
+  public getData(): DataSet {
+    const text = this.getRandomString(this.textLen);
+    const subStringIndex = Math.round(
+      Math.random() * (text.length - this.patternLen)
+    );
+    const pattern = text.substring(
+      subStringIndex,
+      subStringIndex + this.patternLen
+    );
+
+    return new DataSet(
+      this.encoder.encode(text),
+      this.encoder.encode(pattern),
+      text.length
```

Comment:
text.length might be not needed

Reply:
It is needed to verify proper encoded data size on `DataSet` class' initialization. However, we could move this validation earlier. @damiankoper ?

Reply:
I think that since the length parameter does not define the `DataSet`'s state and is not used later on, it is excessive to perform the validation in the constructor. Since it expects already UTF-8 encoded data, we should validate the encoded text before initializing a `DataSet`.

Reply:
Refactored.